28/rewards list

jashpam opened the pull request.

jashpam committed:
chore: selecting different rewards

Teachers are now able to select a specific prize from 4 different types of rewards. UI improvements. Translations to Spanish.

closes #28, 25

juancarlosfarah commented on src/reducers/settings.js:
```diff
@@ -44,6 +46,11 @@ export default (state = INITIAL_STATE, { type, payload }) => {
+    case SET_BADGE_GROUP:
+      return {
+        ...state,
+        badgegroup: payload,
```
You should use camelCase: `badgeGroup`.

juancarlosfarah commented on src/reducers/settings.js:
```diff
@@ -15,6 +16,7 @@ const INITIAL_STATE = {
   subSpaceId: null,
   open: false,
   headerVisible: true,
+  badgegroup: 0,
```
You should use camelCase: `badgeGroup`.

juancarlosfarah commented on src/components/modes/teacher/AssignBadgeForm.js:
```diff
@@ -98,12 +108,14 @@ const mapStateToProps = (state, ownProps) => {
   return {
     studentOptions: state.users.content.map(({ id, name }) => ({ value: id, label: name })),
-    badgeOptions: state.badges.content.map(({ _id, label, color }) => ({
+    badgeOptions: state.badges.groups[state.settings.badgegroup].badges.map(({
```
You should use camelCase: `badgeGroup`.

juancarlosfarah commented on src/components/modes/teacher/Settings.js:
```diff
@@ -80,21 +82,32 @@ class Settings extends Component {
+  handleChangeGroup = (selectedGroup) => {
+    const { value } = selectedGroup;
+    const settingsToChange = {
+      badgegroup: value,
```
You should use camelCase: `badgeGroup`.

juancarlosfarah commented on src/components/modes/teacher/Settings.js:
```diff
@@ -80,21 +82,32 @@ class Settings extends Component {
 
     const {
       headerVisible,
+      badgegroup,
```
You should use camelCase: `badgeGroup`.

juancarlosfarah commented on src/components/modes/teacher/Settings.js:
```diff
@@ -80,21 +82,32 @@ class Settings extends Component {
 
     const { language } = i18n;
     const selectedLanguage = langOptions.find(langOption => langOption.value === language);
+    const selectedGroup = badgeOptions.find(badgeOption => badgeOption.value === badgegroup);
```
You should use camelCase: `badgeGroup`.

juancarlosfarah commented on src/components/modes/teacher/Settings.js:
```diff
@@ -136,9 +164,13 @@ class Settings extends Component {
   settings: PropTypes.shape({
     headerVisible: PropTypes.bool.isRequired,
     lang: PropTypes.string.isRequired,
+    badgegroup: PropTypes.number.isRequired,
```
You should use camelCase: `badgeGroup`.

juancarlosfarah commented on src/components/modes/teacher/Settings.js:
```diff
@@ -148,13 +180,27 @@ Settings.propTypes = {
+    settings: {
+      lang: settings.lang,
+      headerVisible: settings.headerVisible,
+      badgegroup: settings.badgegroup,
```
You should use camelCase: `badgeGroup`.

juancarlosfarah reviewed (changes requested):
Nice commit! I changed it to go to `develop` instead of `master` and have left you a couple minor comments. As soon as you have a look at that, I can merge it and work on deployment.

jashpam committed:
chore: selecting different rewards

Teachers are now able to select a specific prize from 4 different types of rewards. UI improvements. Translations to Spanish.

closes #28, 25

jashpam commented:
Thanks @juancarlosfarah. I have changed 'badgegroup' to 'badgeGroup' in all files now.

jashpam commented:
Thanks @juancarlosfarah. I have changed 'badgegroup' to 'badgeGroup' in all files now.
